src/main/java/com/google/sps/servlets/ChatServlet.java

```diff
@@ -0,0 +1,66 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity; 
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+import com.google.template.soy.SoyFileSet;
+import com.google.template.soy.tofu.SoyTofu;
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/chat")
+public class ChatServlet extends HttpServlet{
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    Query query = new Query("Message").addSort("timestamp", SortDirection.ASCENDING);
+    PreparedQuery pq = datastore.prepare(query);
+
+    ImmutableList.Builder<String> builder = new ImmutableList.Builder<>();
+    for (Entity message : pq.asIterable()) {
+        builder.add((String) message.getProperty("message-text"));
+    }
```

Comment:
Try using a Stream here: https://docs.oracle.com/javase/8/docs/api/java/util/stream/package-summary.html

It'll probably look something like 
```ImmutableList<String> messages = pq.asIterable().stream().map(message -> (String) message.getProperty("message-text")).collect(toImmutableList());```